gallery/src/styles/NxFormLayout/NxFormHorizontalLayoutExample.tsx

```diff
@@ -4,53 +4,84 @@
+            <NxStatefulTextInput validator={validator}/>
           </label>
         </div>
         <div className="nx-form-group">
-          <label className="nx-label">
-            <span className="nx-label__text">Second column</span>
-            <span className="nx-sub-label">This is a sub-label.</span>
-            <input type="text" className="nx-text-input"/>
+          <label className="nx-label nx-label--optional">
+            <span className="nx-label__text">Label</span>
+            <NxStatefulTextInput/>
           </label>
         </div>
-        <div className="nx-form-group">
-          <button className="nx-btn" type="button">Button</button>
-        </div>
-      </div>
-      <div className="nx-form-row">
-        <fieldset className="nx-fieldset">
-          <legend className="nx-label nx-label--optional">Checkboxes</legend>
-          <NxCheckbox isChecked={false}>Checkbox 1</NxCheckbox>
-          <NxCheckbox isChecked={true}>Checkbox 2</NxCheckbox>
-          <NxCheckbox isChecked={false}>Checkbox 3</NxCheckbox>
-        </fieldset>
-        <fieldset className="nx-fieldset">
-          <legend className="nx-label">Radio buttons</legend>
-          <NxRadio name="demo1" value="demo1" isChecked={true}>Radio Button 1</NxRadio>
-          <NxRadio name="demo2" value="demo2" isChecked={false}>Radio Button 2</NxRadio>
-          <NxRadio name="demo3" value="demo3" isChecked={true}>Radio Button 3</NxRadio>
-        </fieldset>
       </div>
+      <fieldset className="nx-fieldset">
+        <legend className="nx-legend">
+          <span className="nx-legend__text">
+            Checkboxes
+          </span>
+        </legend>
+        <NxCheckbox onChange={toggleRed} isChecked={isRed}>Red</NxCheckbox>
+        <NxCheckbox onChange={toggleBlue} isChecked={isBlue}>Blue</NxCheckbox>
+        <NxCheckbox onChange={toggleGreen} isChecked={isGreen}>Green</NxCheckbox>
+      </fieldset>
+      <fieldset className="nx-fieldset">
+        <legend className="nx-legend"><span className="nx-legend__text">Radio buttons</span></legend>
+        <NxRadio name="color"
+                 value="red"
+                 onChange={setColor}
+                 isChecked={color === 'red'}
+                 radioId="color-red">
+          Red
+        </NxRadio>
+        <NxRadio name="color"
+                 value="purple"
+                 onChange={setColor}
+                 isChecked={color === 'purple'}
+                 radioId="color-purple">
+          Purple
+        </NxRadio>
+        <NxRadio name="color" value="green" onChange={setColor} isChecked={color === 'green'} radioId="color-green">
+          Green
+        </NxRadio>
+        <NxRadio name="color" value="blue" onChange={setColor} isChecked={color === 'blue'} radioId="color-blue">
+          Blue
+        </NxRadio>
+      </fieldset>
+      <footer className="nx-form-footer">
+        <NxErrorAlert>This is an example <strong>error</strong> message.</NxErrorAlert>
```

Comment:
What scenario are we trying to emulate here, and what pattern are we trying to demonstrate for dealing with it?  In particular, an error like this would typically appear when there is an error saving the form.  I know @dturner32 likes to always have Retry buttons for things like that, so do we need one here?  Should we be demonstrating to the user that `NxLoadError` would actually be the most correct component for handling that situation, rather than a bare `NxErrorAlert`?  If there is a Retry button, should there also still be the Submit button?  If so would those two buttons do the same thing?

Reply:
> What scenario are we trying to emulate here, and what pattern are we trying to demonstrate for dealing with it? In particular, an error like this would typically appear when there is an error saving the form. I know @dturner32 likes to always have Retry buttons for things like that, so do we need one here? Should we be demonstrating to the user that `NxLoadError` would actually be the most correct component for handling that situation, rather than a bare `NxErrorAlert`? If there is a Retry button, should there also still be the Submit button? If so would those two buttons do the same thing?

Because this is supposed to be a layout example I was attempting to demonstrate how NxAlert lays out in a form footer. It doesn't have to be an error alert we could demonstrate a different kind of alert.

Reply:
> It doesn't have to be an error alert we could demonstrate a different kind of alert.

Let's do that.  I wouldn't want people thinking this is the preferred way to communicate errors when it is missing pieces.  Then there's the question of whether Design thinks other sorts of alerts should go in the footer in the first place. @dturner32 ?

Reply:
> > It doesn't have to be an error alert we could demonstrate a different kind of alert.
> 
> Let's do that. I wouldn't want people thinking this is the preferred way to communicate errors when it is missing pieces. Then there's the question of whether Design thinks other sorts of alerts should go in the footer in the first place. @dturner32 ?

NxInfoAlert: https://github.com/sonatype/sonatype-react-shared-components/pull/122/commits/085ddd03220d314b9bfd2ec3435f1f6e5745ef08